wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java

```diff
@@ -760,6 +763,20 @@ private String replaceSnapshot(MachineImpl machine, String namespace) {
         }
     }
 
+    private void updateWorkspaceSnapshots(String id, Workspace workspace, Workspace update) throws SnapshotException {
+        if (!workspace.getConfig().getDefaultEnv().equals(update.getConfig().getDefaultEnv())) {
+            for (SnapshotImpl snapshot : snapshotDao.findSnapshots(id)) {
+                try {
+                    snapshotDao.removeSnapshot(snapshot.getId());
```

Comment:
I think it should be replaced with update of snapshot. I know that snapshot dao miss that method